wave_utils/src/config.cpp

```diff
@@ -0,0 +1,362 @@
+#include "wave/utils/config.hpp"
+
+
+namespace wave {
+
+ConfigParser::ConfigParser(void) {
+    this->configured = false;
+    this->loaded = false;
+}
+
+int ConfigParser::getYamlNode(std::string key, YAML::Node &node) {
+    int end;
+    std::string s;
+    std::vector<std::string> elements;
+    std::vector<YAML::Node> traversal;
+
+    // split key into elements
+    s = "";
+    for (int i = 0; i < key.length(); i++) {
```

Comment:
Done: 

```
int ConfigParser::getYamlNode(std::string key, YAML::Node &node) {
    std::string element;
    std::istringstream iss(key);
    std::vector<YAML::Node> traversal;

    // recurse down config key
    traversal.push_back(this->root);
    while (std::getline(iss, element, '.')) {
        traversal.push_back(traversal.back()[element]);
    }
    node = traversal.back();
    // Note:
    //
    //    yaml_node = yaml_node["some_level_deeper"];
    //
    // Destroys the parsed yaml tree/graph, to avoid this problem
    // we store the YAML::Node traversed into a std::vector
    // and return the last visited YAML::Node

    return 0;
}
```